problems/longest_substring_w-o_repeats.py

```diff
@@ -0,0 +1,32 @@
+#https://leetcode.com/submissions/detail/508592080/
+
+class Solution(object):
+    def lengthOfLongestSubstring(self, s):
+        """
+        :type s: str
+        :rtype: int
+        """
+        if len(s) == 0:
+            return 0
+        
+        i = 0
+        word = ''
+        maxlen = 0
+        
+        while i < len(s):
```

Comment:
this is O(n^2) time, right? we can do better. think about the repeated work being done with this nested loop.

Reply:
<details>
  <summary>Hint</summary>
  
  you're building the same substrings of each longest unique string sequence
  E.g. abcdd
  first run builds abcd
  second run builds bcd
  third run builds cd

  see the wasted work?
</details>